Defer Device Screen Dimensions gathering to SideMenu construction

thejettdurham opened the pull request:
Fixes #323.

This may also help with the issues mentioned in #153, but its not clear to me what behavior is desired when the device is rotated with an open menu.

This resolved an issue I was experiencing in my app, and doesn't appear to have broken anything in the provided example app.

thejettdurham committed:
Defer Device Screen Dimensions gathering to SideMenu construction

Kureev commented on index.js:
```diff
@@ -1,18 +1,10 @@
 import PropTypes from 'prop-types';
 import styles from './styles';
 
-type WindowDimensions = { width: number, height: number };
```
Any reasons to remove it?

thejettdurham replied:
Not sure why I did this, but I added it back in.

Kureev commented on index.js:
```diff
@@ -71,12 +62,14 @@ export default class SideMenu extends React.Component {
   constructor(props: Props) {
     super(props);
 
+    const { width, height } = Dimensions.get('window');
```
it can be typed using `WindowDimensions`

thejettdurham replied:
Fixed

Kureev commented on index.js:
```diff
@@ -275,13 +269,15 @@ SideMenu.propTypes = {
   autoClosing: PropTypes.bool,
 };
 
+const initialDeviceScreen = Dimensions.get('window');
```
You are not using this variable anywhere else but in the `defaultProps`. I'd rather recommend you to use `Dimensions.get('window').width` inside `defaultProps` directly

thejettdurham replied:
Fixed

Kureev commented on index.js:
```diff
@@ -91,13 +84,14 @@ export default class SideMenu extends React.Component {
-      hiddenMenuOffset: deviceScreen.width * hiddenMenuOffsetPercentage,
+      hiddenMenuOffset: width * hiddenMenuOffsetPercentage,
       left,
+      barrierForward: width / 4,
```
I don't see you update this value anywhere. Why do you want to keep it inside the state?

thejettdurham replied:
`this.state.barrierForward` is used in `SideMenu.handlePanResponderEnd`, passed to the module level function `shouldOpenMenu`. Before, `shouldOpenMenu` didn't take `barrierForward` as an argument, instead using a value calculated from the module's scope. 

The old module-level `barrierForward` variable is calculated based on `deviceScreen`, which is never updated to account for device rotations. Since this PR removes the module-level `deviceScreen` in favor of getting fresh device dimensions in the component's constructor, I had to change a few other things that were using this value, `shouldOpenMenu` being one of them

Kureev reviewed (changes requested):
Hey @thejettdurham!

Thanks for composing this PR and sorry for taking ages to review it. It's pretty hard to keep repos maintained once you're not actively working with the component.

I added some questions and nitpicks to your PR. If you're not interested in this PR just let me know and I'll apply them on my own. Otherwise, it would be wonderful if you can fix them.

Let me know anyway and we proceed with this PR.

thejettdurham commented:
@Kureev Sorry for letting this go so long! My schedule got really busy around the end of Feb and is just now starting to open back up. I'm addressing your comments on this PR now, and hope to have them cleared up quickly 😄

thejettdurham commented on index.js:
```diff
@@ -1,18 +1,10 @@
-  Animated,
-  TouchableWithoutFeedback,
-} from 'react-native';
+import {Animated, Dimensions, PanResponder, TouchableWithoutFeedback, View} from 'react-native';
```
Sorry about this, I have Prettier hooked up in my editor, having it set to format and optimize imports on save. I'll be more careful about this in the future ;)

thejettdurham committed:
Addresses issues raised in PR

nferrara100 commented:
Would love to see this merged!

thejettdurham commented:
Well, as long as this PR has been open, I don't see any hope for it getting merged in its current state. Even so, I've long-since left the job where I had this issue, so I'm not even in a position to re-verify these changes.

With that, I'm going to close this PR and the associated issue, leaving it up to the maintainers whether they want to see this change added.
